README.md

```diff
@@ -1,18 +1,40 @@
-# Convex
+This is a chess app built on Convex.
 
-This example demonstrates how to build a fully functional chat app using React, Next.js and Convex.
+## Testing
 
-The app was build within few hours. The computer just does random valid moves for now. Will improve later.
+This app has tests that run against a local Convex backend.
 
-## Run locally
+To run these tests:
 
-```bash
-npm install
-rm -rf convex.json
-npx convex init
-npm run dev
-```
+- Clone the Convex open source backend repo and follow setup instructions
+- Set `CONVEX_LOCAL_BACKEND` to the path of the repo
+- `npm run test`
 
-## Deploy your own
+How this works:
 
-Deploy the example using [Vercel](https://docs.convex.dev/using/hosting/vercel):
+- `npm run test` ensures a Convex local backend is running (`testing/checkLocalBackend.js`)
+- It then pushes the functions in this repo to the backend
+- It sets the `IS_TEST` environment variable, which allows running "test only" functions (`convex/testing/`)
+- It runs jest tests using the `ConvexClient` hooked up to the local backend (`testing/ConvexTestingHelper.ts`)
+
+`tests.test.ts` has some examples of tests.
```

Comment:
tests.test.ts is a bit odd, if folks don't know the `.test.ts` pattern. could it be `example.test.ts`?